fpx/src/api/handlers/canned_requests.rs

```diff
@@ -0,0 +1,120 @@
+use crate::api::errors::{AnyhowError, ApiServerError};
+use crate::api::Config;
+use crate::canned_requests::{CannedRequest, SaveLocation};
+use axum::extract::State;
+use axum::response::{IntoResponse, Response};
+use axum::Json;
+use http::StatusCode;
+use serde::{Deserialize, Serialize};
+use std::collections::BTreeMap;
+use thiserror::Error;
+use tracing::{error, instrument};
+
+#[instrument(skip(config))]
+pub async fn canned_request_create(
+    State(config): State<Config>,
+    Json(mut canned_request): Json<NewCannedRequest>,
+) -> Result<StatusCode, ApiServerError<CannedRequestCreateError>> {
+    let save_location = SaveLocation::try_parse(&canned_request.type_, config.fpx_directory)
+        .map_err(|_| CannedRequestCreateError::InvalidType)?;
+
+    canned_request.request.name = canned_request.name;
+    canned_request
+        .request
+        .save(save_location)
+        .await
+        .map_err(|err| CannedRequestCreateError::Internal(err.into()))?;
+
+    Ok(StatusCode::CREATED)
+}
+
+#[derive(Debug, Deserialize)]
+#[allow(dead_code)]
+struct NewCannedRequest {
+    #[serde(rename = "type")]
+    type_: String,
+    name: String,
+    request: CannedRequest,
+}
+
+#[derive(Debug, Error, Serialize, Deserialize)]
+#[serde(tag = "error", content = "details", rename_all = "camelCase")]
+#[non_exhaustive]
+#[allow(dead_code)]
+pub enum CannedRequestCreateError {
+    #[error("unknown type, expected `ephemeral`, `personal` or `shared`")]
+    InvalidType,
+
+    #[error("failed to handle request: {0:?}")]
+    Internal(
```

Comment:
I don't think we should have an `Internal` variant, given that we already have the CommonErrors Internal error. 

Regarding the easy propagation using the `?` operator, I think we should investigate where we are returning these anyhow errors and either refactor them to a typed error or just do a `map_error`

I think we might be able to create a `From` for anyhow Error into a `ApiServerError<E>`, which should always result in a Common error. I will investigate this shortly.